tools/spm/generate_partition_code.py

```diff
@@ -0,0 +1,439 @@
+#!/usr/bin/python
+import glob
+import itertools
+import json
+import os
+from os.path import join as path_join
+
+from jinja2 import Environment, FileSystemLoader
+from jsonschema import validate
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+TEMPLATES_DIR = path_join(SCRIPT_DIR, 'templates')
+
+
+def assert_int(num):
+    """
+    Tries to parse an integer num from a given string
+    
+    :param num: Number in int/string type
+    :return: Numeric value
+    """
+    if isinstance(num, int):
+        return num
+    num_str = str(num)
+    radix = 16 if num_str.lower().startswith('0x') else 10
+    res = int(num_str, radix)
+    # Python converts str to int as a signed integer
+    if res > 0x7FFFFFFF:
+        res -= 0x100000000
+    return res
+
+
+class SecureFunction(object):
```

Comment:
This class is a giant tuple. This could be written as a named tuple and a generator function:
```python
from collections import namedtuple

SEC_FUNC = namedtuple("SecureFunction", 
                      "sfid signal non_secure_clients " 
                      "minor_version minor_policy")

def SecureFunction(sfid, signal, non_secure_clients, minor_version=1, minor_policy='strict'):
     assert isinstance(non_secure_clients, bool),\
         'non_secure_clients parameter must be of boolean type' 
    minor_version = assert_int(minor_version)
    assert self.minor_version > 0, 'minor_version parameter is invalid'
    assert minor_policy in self.MINOR_POLICIES, 'minor_policy parameter is invalid'
    return SEC_FUNC(sfid, signal, non_secure_clients, minor_version, minor_policy)
```
This has the benefit/downside that you can't overwrite the fields of the object, and the __eq__ method is written for you by python.

Reply:
I should point out that this comment is not blocking.